message.go

```diff
@@ -19,6 +21,12 @@ type Message struct {
 	// If not set at the creation, Time will be automatically set when
 	// writing the message.
 	Time time.Time
+
+	// Compression codec used to encode the message value
+	CompressionCodec int8
+
+	// Compression level for the codec if supported (only gzip)
+	CompressionLevel int
```

Comment:
Should those field be changed to a `CompressionCodec` instead? Having the compression level part of the message here is really mixing abstraction layers that should be decoupled. For example, of we needed to support new compression options, should they all be carried in the message as well?